client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -185,6 +193,30 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        ReaderGroupConfig oldConfig = getReaderGroupConfig();
+        Set<Stream> newStreams = config.getStartingStreamCuts().keySet();
+        Set<Stream> oldStreams = oldConfig.getStartingStreamCuts().keySet();
+
+        Set<Stream> streamsToSub = Sets.difference(config.isSubscriberForRetention() ? newStreams : Collections.emptySet(),
```

Comment:
can we refactor this to a different method and simplify the logic here?

Reply:
Cleaned up the `resetReaderGroup` as suggested by moving this logic to a new private method called `manageSubscriptions`.